In [1]:
from astropy.table import QTable
import astropy.units as u
import numpy as np


In [26]:
names = ('Center Rest Frequency', 
         'Molecule and Transition',
        'Eff. Ch. #', 
        'Velocity Bandwidth',
        'Velocity Resolution')
freq = [230.538,    # 12 CO
       231.900928,  # H30
       218.76006,   # H2CO 322-221
       218.324723,  # HC3N
       218.222192,  # H2CO 303-202
       217.104980,  # SiO 5-4
       220.398684,  # 13CO
       219.560358  # C18O
       ] * u.GHz

spw = [ '\ce{^12CO} $J=2\shortrightarrow1$',
        'H(30)$\alpha$',
        '\ce{H2CO} $J=3_{2,2}\shortrightarrow2_{2,1}$',
        '\ce{HC3N} v=0 $J=24\shortrightarrow23$',
        '\ce{H2CO} $J=3_{0,3}\shortrightarrow2_{0,2}$',
        'SiO v=0 $J=5\shortrightarrow4$',
        '\ce{^13CO} $J=2\shortrightarrow1$',
        '\ce{C^18O} $J=2\shortrightarrow1$']

ch = [3840, #12 CO  
     3840,  #H30   
     960,   # H2CO 322-221   
     960,   # HC3N   
     960,   # H2CO 303-202     
     960,   # SiO 5-4 
     1920,  # 13CO  
     1920]  # C18O  

band = [304.8, #12 CO
       2424.2, #H30
       321.2,  # H2CO 322-221  
       321.9,  # HC3N  
       322.0,  # H2CO 303-202  
       323.7,  # SiO 5-4  
       318.8,  # 13CO  
       320.1   #C18O
       ]*u.km/u.second

reso = [0.159, #12 CO 
       1.263,  #H30   
       0.774,  # H2CO 322-221   
       0.671,  # HC3N    
       0.671,  # H2CO 303-202   
       0.674,  # SiO 5-4  
       0.332,  # 13CO  
       0.333   # C18O  
        ] *u.km/u.second
    

In [18]:
len(freq)

8

In [36]:
# 12CO
row1 = [freq[0], spw[0], ch[0], band[0], reso[0]]
row2 = [freq[1], spw[1], ch[1], band[1], reso[1]]

In [52]:
#a = np.array([1, 4, 5], dtype=np.int32)
#b = [2.0, 5.0, 8.5]
#c = ['x', 'y', 'z']
#d = [10, 20, 30] * u.m / u.s

t = QTable([freq[:3], spw[:3], ch[:3], band[:3], reso[:3]],
           names=names,
           meta={'name': 'first table'})

In [53]:
# OCS
t.add_row([218.903355*u.GHz, '\ce{OCS} $J=18\shortrightarrow17$', 960, 321.2*u.km/u.second, 0.774*u.km/u.second])

In [54]:
# HC3N
t.add_row([freq[3], spw[3], ch[3], band[3], reso[3]])

In [55]:
# CH3OH
t.add_row([218.44005*u.GHz, '\ce{CH3OH} $J=4_{2,2}\shortrightarrow3_{1,2}$', 960, 321.9*u.km/u.second, 0.671*u.km/u.second])

In [56]:
# extra H2CO
t.add_row([218.475632*u.GHz, '\ce{p-H2CO} $J=3_{22}\shortrightarrow2_{21}$', 960, 321.9*u.km/u.second, 0.671*u.km/u.second])

In [57]:
# H2CO 303-202
t.add_row([freq[4], spw[4], ch[4], band[4], reso[4]])
# SiO 5-4
t.add_row([freq[5], spw[5], ch[5], band[5], reso[5]])
# 13CO
t.add_row([freq[6], spw[6], ch[6], band[6], reso[6]])
# C18O
t.add_row([freq[7], spw[7], ch[7], band[7], reso[7]])


In [60]:
from astropy.io import ascii


In [61]:
ascii.write(t, format='latex')

\begin{table}
\begin{tabular}{ccccc}
Center Rest Frequency & Molecule and Transition & Eff. Ch. # & Velocity Bandwidth & Velocity Resolution \\
$\mathrm{GHz}$ &  &  & $\mathrm{km\,s^{-1}}$ & $\mathrm{km\,s^{-1}}$ \\
230.538 & \ce{^12CO} $J=2\shortrightarrow1$ & 3840 & 304.8 & 0.159 \\
231.900928 & H(30)$lpha$ & 3840 & 2424.2 & 1.263 \\
218.76006 & \ce{H2CO} $J=3_{2,2}\shortrightarrow2_{2,1}$ & 960 & 321.2 & 0.774 \\
218.903355 & \ce{OCS} $J=18\shortrightarrow17$ & 960 & 321.2 & 0.774 \\
218.324723 & \ce{HC3N} v=0 $J=24\shortrightarrow23$ & 960 & 321.9 & 0.671 \\
218.44005 & \ce{CH3OH} $J=4_{2,2}\shortrightarrow3_{1,2}$ & 960 & 321.9 & 0.671 \\
218.475632 & \ce{p-H2CO} $J=3_{22}\shortrightarrow2_{21}$ & 960 & 321.9 & 0.671 \\
218.222192 & \ce{H2CO} $J=3_{0,3}\shortrightarrow2_{0,2}$ & 960 & 322.0 & 0.671 \\
217.10498 & SiO v=0 $J=5\shortrightarrow4$ & 960 & 323.7 & 0.674 \\
220.398684 & \ce{^13CO} $J=2\shortrightarrow1$ & 1920 & 318.8 & 0.332 \\
219.560358 & \ce{C^18O} $J=2\shortrighta

\begin{table*}[ht!]
\centering
\caption{Correlator Configuration - The observation parameters of each spectral window. 
 \label{test table}}

\begin{tabular}{|l|l|l|l|l|l|} 
\hline
\label{tab:obs}
\centering
\begin{tabular}[c]{@{}l@{}}Center Rest \\Freq (GHz)\end{tabular} & Spectral Window Name           & Eff. Ch \# & \begin{tabular}[c]{@{}l@{}}Vel. Bandwidth\\(km/s)\end{tabular} & \begin{tabular}[c]{@{}l@{}}Vel. Resolution\\(km/s)\end{tabular}  \\ 
\hline
230.538000                                                       & CO 2-1         & 3840       & 304.8                                                          & 0.159                                                            \\ 
\hline
231.900928                                                       & H(30)              & 3840       & 2424.2                                                         & 1.263                                                            \\ 
\hline
218.760066                                                       & H$_2$CO 3(2,2)-2(2,1) & 960        & 321.2                                                          & 0.774                                                            \\ 
\hline
218.324723                                                       & HC$_3$N v=0 J=24-23   & 960        & 321.9                                                          & 0.671                                                            \\ 
\hline
218.222192                                                       & H$_2$CO 3(0,3)-2(0,2) & 960        & 322.0                                                          & 0.671                                                            \\ 
\hline
217.104980                                                       & SiO v=0 5-4        & 960        & 323.7                                                          & 0.674                                                            \\ 
\hline
220.398684                                                       & 13CO 2-1       & 1920       & 318.8                                                          & 0.332                                                            \\ 
\hline
219.560358                                                       & C18O 2-1           & 1920       & 320.1                                                          & 0.333                                                            \\
\hline
\end{tabular}

\end{table*}